qa/tasks/ceph.py

```diff
@@ -1137,6 +1137,43 @@ def first_in_ceph_log(pattern, excludes):
                             )
                         break
 
+        def first_in_kern_log(pattern, excludes):
+            """
+            Find the first occurrence of the pattern specified in the kern log,
+            Returns None if none found.
+
+            :param pattern: Pattern scanned for.
+            :param excludes: Patterns to ignore.
+            :return: First line of text (or None if not found)
+            """
+            args = [
+                'sudo',
+                'egrep', pattern,
+                '/var/log/kern.log',
+            ]
```

Comment:
Note that you can also use journalctl here, for most modern machines. Something like:
```
journalctl -k --since today -g <pattern>
```

Reply:
I came across this option and chose the old way for no specific reason. Also found few distributions compile journalctl without pattern matching capability.

$ journalctl -k --since today --grep='BUG:|Oops'
Compiled without pattern matching support
$ uname -a
Linux teuthology 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

Reply:
I had triggered a test run for this and found that the '/var/log/kern.log' missing. 

...
2021-07-08T11:01:00.838 INFO:teuthology.orchestra.run.smithi068.stderr:grep: /var/log/kern.log: No such file or directory
2021-07-08T11:01:00.843 INFO:tasks.ceph:Unmounting /var/lib/ceph/osd/ceph-0 on ubuntu@smithi068.front.sepia.ceph.com
...

Is it not guaranteed to present in all systems? So the option is to run 'journalctl -k --since today' and pipe it to grep ?

Reply:
Yeah, different distros put logs in different places. journalctl should be available on most of them though. Piping the output from that into grep sounds fine.

Reply:
Done